File: java/JavaFileutil/src/JavaFileutil/JavaFileCommands/MoveCommand.java
package JavaFileutil.JavaFileCommands;

import JavaFileutil.AbstractCommands.AbstractBinaryFileCommand;

import java.nio.file.Files;
import java.util.List;

public class MoveCommand extends AbstractBinaryFileCommand
{
    public static final String UsageString =
        "move command moves <file1> to the location specified in <file2>.\n" +
        "Use it the following way:\n" +
        "\t'move <file1> to <file2>'\n" +
        "\t" + NotePreface + ArgumentCountMismatchMessage;

    public MoveCommand(List<String> args)
    {
        super(args);
    }

    @Override
    public void Execute()
    {
        if (!Validation.Validate())
            return;

        try
        {
            Files.move(Source.toPath(), Destination.toPath());
        }
        catch (Exception e)
        {
            System.out.format("Move exception.\nError message: %s.", e.getMessage());
        }
    }
}

File: python/PythonFileCommands/MoveCommand.py
from .CopyCommand import *
from shutil import move


class MoveCommand(CopyCommand):
    @classmethod
    def usage_string(cls):
        return "move command moves <file1> to the location specified in <file2>.\n" + \
               "Use it the following way:\n" + \
               "\t'move <file1> to <file2>'\n" + \
               "\t" + cls._note_preface() + cls._argument_count_mismatch_message()

    def __init__(self, args):
        super().__init__(args)

    def execute(self):
        if not self._validation.validate():
            return
        try:
            move(str(self._source), str(self._destination))
        except IOError as ioe:
            print("Move exception.\nError: {}".format(ioe))
